tests/test_modeling_common.py

```diff
@@ -4532,6 +4532,8 @@ def test_flash_attention_2_padding_matches_padding_free_with_position_ids(self):
                 config.max_position_embeddings = max_new_tokens + dummy_input.shape[1] + 1
 
             model = model_class(config)
+            if "position_ids" not in inspect.signature(model.forward).parameters:
+                self.skipTest("Model does not support position_ids")
```

Comment:
For the record: `self.skipTest` will skip here and break the for loop `for model_class in self.all_generative_model_classes`.

Here is somehow fine as it's rare to have a model architecture with some model classes having `position_ids` but not other model class.

Reply:
That is indeed undesirable :(

Do you have a better suggestion? `continue` to skip the iteration on the loop also doesn't seem adequate.

Reply:
we can have `pytest-subtests` installed in CI env. and using `TestCase.subTest` and skip inside it where necessary.

But I have installed it yet because it's will change the test summary outputs (for which some adjustments have to be done IMO).

(I also had to fix something in [pytest-subtests](https://github.com/pytest-dev/pytest-subtests/pull/169) repository)

Reply:
For now, let's just do whatever you prefer (skip or continue), it's fine, we are aware of the situation and might take an action.

Reply:
Okay :) 

I'm going to merge as is then -- as you wrote, it's uncommon for a model to have subclasses with different inputs regarding the input modality